### DISCUSSION TITLE
Error: P1001: Can't reach database server

### DISCUSSION THREAD

**Discussion by mallusrgreatv2** [prisma/prisma#22580] [Q&A]
### Question

My db connection string is currently: `mysql://(user):(pass)@(host):3306/test`
I get the error in the title. 
In the mariadb server: [Warning] Aborted connection 464968 to db: 'unconnected' user: 'unauthenticated' host: 'my ip' (This connection closed normally without authentication)
I have already given the privileges needed.
![image](https://github.com/prisma/prisma/assets/69511006/2844f2c6-d6a8-4f73-9e22-b05c75fa7ed0)

### Information about Prisma Schema, Client Queries and Environment (optional)
```prisma
generator client {
    provider = "prisma-client-js"
}

datasource db {
    provider = "mysql"
    url      = env("DATABASE_URL")
}

model User {
    id                    String    @id @default(cuid())
    name                  String?
    email                 String?   @unique
    emailVerified         DateTime?
    image                 String?
    accounts              Account[]
    sessions              Session[]
    roles                 Role[]
    freeGeneratedCount    Int       @default(0)
    basicGeneratedCount   Int       @default(0)
    premiumGeneratedCount Int       @default(0)
    opGeneratedCount      Int       @default(0)
}

model Role {
    name   String  @unique
    User   User?   @relation(fields: [userId], references: [id])
    userId String?
}

model Session {
    id           String   @id @default(cuid())
    sessionToken String   @unique
    userId       String
    expires      DateTime
    user         User     @relation(fields: [userId], references: [id], onDelete: Cascade)
}

model Account {
    id                String  @id @default(cuid())
    userId            String
    type              String
    provider          String
    providerAccountId String
    refresh_token     String? @db.Text
    access_token      String? @db.Text
    expires_at        Int?
    token_type        String?
    scope             String?
    id_token          String? @db.Text
    session_state     String?

    user User @relation(fields: [userId], references: [id], onDelete: Cascade)

    @@unique([provider, providerAccountId])
}

**Comment by ludralph** [comment]
Hi @mallusrgreatv2 👋 

Can you please confirm that you can access the database using a Database Viewer like [TablePlus](https://tableplus.com/)? Also, can you try the solution suggested [here](https://dba.stackexchange.com/a/320590)?

**Reply by mallusrgreatv2** [reply]
For the first one, the image included is directly from TablePlus. For the second one, I am not interested in suppressing it. That error is related to the issue I'm having.

**Reply by ludralph** [reply]
Hi @mallusrgreatv2 

Does your mariadb server require ssl/tls connection? If yes, you may need to include the necessary SSL parameters. For example: `mysql://(user):(pass)@(host):3306/test?ssl=true&sslmode=verify-full`

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!;
